person.rb

```diff
@@ -20,4 +21,48 @@ def of_age?
   def can_use_services?
     of_age? || @parent_permission
   end
+
+  def correct_name
+    @name
+  end
+end
+
+# Nameable class
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
 end
+
+# Base decorator
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+# Capitalize decorator
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+# Trimmer decorator
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.length <= 10 ? @nameable.correct_name : @nameable.correct_name[0..9]
+  end
+end
```

Comment:
- [ ] [OPTIONAL] It would be better to include these classes in their own individual .rb files instead of being put in the person.rb file all together.